WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsDataSource.swift

```diff
@@ -15,6 +15,10 @@ final class RefundDetailsDataSource: NSObject {
     ///
     private let order: Order
 
+    /// Refund Detail View Model
+    ///
+    private let viewModel: RefundDetailsViewModel
```

Comment:
it looks like the view model `RefundDetailsViewModel` also has a reference to the data source: https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/ViewModels/RefundDetails/RefundDetailsViewModel.swift#L37-L39 - there is no retain cycle because a new instance of view model was created in `init`, but maybe we could separate the responsibilities for these two classes so that they don't have to keep a reference to each other? It seems like currently the data source `RefundDetailsDataSource` is the table view data source, wrapped in the view model. I don't understand fully why the view model owns the data source, my understanding of view models is to supply UI data from model data (like `refund` and `order` in this case). The diffs in the data source and view model in this PR look great to me, I just wasn't sure about the inter-dependency from the setup before this PR where the view model owns the data source 🤔

Reply:
🤦‍♀  I'm not sure why I added a circular reference. Fixing now

Reply:
Edit: okay, so I remember what I was thinking. Some cells pass in a viewModel so they can be configured privately inside the cell. I must have noticed the pattern of passing in the RefundDetailsViewModel to a couple of cells and refactored it out to use a single property in the class instead, but without considering which class I was in. The roles of the view model and the data source aren't completely clear to me either.